sunpy/coordinates/frames.py

```diff
@@ -474,6 +475,16 @@ def make_3d(self):
 
         rep = self.represent_as(UnitSphericalRepresentation)
         lat, lon = rep.lat, rep.lon
+
+        lon_dtype = getattr(lon, 'dtype', float)
+        lat_dtype = getattr(lat, 'dtype', float)
```

Comment:
Doesn't 'float' default to the system precision? So this check would do the reverse on a native 32bit machine? Would it not be better to check explicitly for `np.float64`?

Reply:
`float` is defined by the Python implementation.  In CPython, it's always a C `double`, so it's 64-bit even on 32-bit machines.  However, there could certainly be Python implementations that don't make the same choice, so I'll tweak this.

Reply:
Actually, on further thought, if the user is on a machine where the Python's built-in `float` is only 32 bits – and I don't know what Python implementations even do that – I'm not sure we should raise any warning that they're not using 64-bit floats...